bootstore/src/schemes/v0/request_manager.rs

```diff
@@ -0,0 +1,401 @@
+    /// Create a new RequestManager
+    pub fn new(id: Baseboard, config: Config) -> RequestManager {
+        RequestManager {
+            id,
+            config,
+            requests: BTreeMap::new(),
+            expiry_to_id: BTreeMap::new(),
+            envelopes: vec![],
+        }
+    }
+
+    /// Return an iterator of drained envelopes
+    pub fn drain_elements(&mut self) -> impl Iterator<Item = Envelope> + '_ {
+        self.envelopes.drain(..)
+    }
+
+    /// Track a new `Fsm::init_rack` api request and broadcast a
+    /// `RequestType::Init` to connected peers.
+    pub fn new_init_rack_req(
+        &mut self,
+        now: Instant,
+        rack_uuid: Uuid,
+        packages: BTreeMap<Baseboard, SharePkg>,
+        connected_peers: &BTreeSet<Baseboard>,
+    ) -> Uuid {
+        let expiry = now + self.config.rack_init_timeout;
+        let mut acks = InitAcks::default();
+        acks.expected = packages
+            .keys()
+            .cloned()
+            .filter_map(|id| if id != self.id { Some(id) } else { None })
+            .collect();
+        let req = TrackableRequest::InitRack {
+            rack_uuid,
+            packages: packages.clone(),
+            acks,
+        };
+        let request_id = self.new_request(expiry, req);
+
+        // Send a `Request::Init` to each connected peer in the initial group
+        let iter = packages.into_iter().filter_map(|(to, pkg)| {
+            if connected_peers.contains(&to) {
+                Some(Envelope {
+                    to,
+                    msg: Request {
+                        id: request_id,
+                        type_: RequestType::Init(pkg),
+                    }
+                    .into(),
+                })
+            } else {
+                None
+            }
+        });
+        self.envelopes.extend(iter);
+        request_id
+    }
+
+    /// Track a new `Fsm::load_rack_secret` api request and broadcast a
+    /// `RequestType::GetShare` to connected peers.
+    pub fn new_load_rack_secret_req(
+        &mut self,
+        now: Instant,
+        rack_uuid: Uuid,
+        threshold: u8,
+        connected_peers: &BTreeSet<Baseboard>,
+    ) -> Uuid {
+        let expiry = now + self.config.rack_secret_request_timeout;
+        let request_id = self.new_request(
+            expiry,
+            TrackableRequest::LoadRackSecret {
+                rack_uuid,
+                acks: ShareAcks::new(threshold),
+            },
+        );
+        self.broadcast_get_share(request_id, rack_uuid, connected_peers);
+        request_id
+    }
+
+    /// Track a new `RequestType::Learn` request received from a peer and
+    /// broadcast to all connected peers to get shares to unlock the rack
+    /// secret, so the "extra" shares can be decrypted and on can be handed out
+    /// to the learning peer.
+    pub fn new_learn_received_req(
+        &mut self,
+        request_id: Uuid,
+        now: Instant,
+        rack_uuid: Uuid,
```

Comment:
Seeing these two `Uuid` next to each other makes me wonder if we want a wrapper `struct RackUuid(Uuid)` to avoid the possibility of mix-ups.

Reply:
Done in [1883147](https://github.com/oxidecomputer/omicron/pull/3385/commits/188314726f7cc022ed06afa42ad9be27eee5364e)